IConfiguration should not use ts.CompilerOptions

### Summary
  - **SDK Version**: 3.0.8
  - **Expected behavior**: accept `compilerOptions` as the same interface used in `tsconfig.json`
  - **Actual behavior**: the type `TsConfig`, which is internal usage only, is not compatible with `CompilerOptions`

### Code occuring the bug
```typescript
import type nestia from 'nestia';

export const NESTIA_CONFIG: nestia.IConfiguration = {
  input: [
    './src/app/*.controller.ts',
    './src/modules/**/*.controller.ts',
  ],
  output: './src/sdk',
  json: true,
  swagger: {  output: './src/assets/api-docs.json' },
  compilerOptions: {
    module: 'commonjs',
  },
};

export default NESTIA_CONFIG;
```

_error_
```
error TS5024: Compiler option 'module' requires a value of type string.
```

--- a/src/IConfiguration.ts
+++ b/src/IConfiguration.ts
@@ -1,4 +1,5 @@
 import ts from "typescript";
+import type { StripEnums } from "./utils/StripEnums";
 
 /**
  * Definition for the `nestia.config.ts` file.
@@ -38,7 +39,7 @@ export interface IConfiguration {
      * }
      * ```
      */
-    compilerOptions?: ts.CompilerOptions;
+    compilerOptions?: StripEnums<ts.CompilerOptions>;
 
     /**
      * Whether to assert parameter types or not.

--- a/src/utils/StripEnums.ts
+++ b/src/utils/StripEnums.ts
@@ -1,0 +1,10 @@
+export type StripEnums<T extends Record<string, any>> = {
+    [Key in keyof T]: T[Key] extends
+        | string
+        | boolean
+        | object
+        | undefined
+        | any[]
+        ? T[Key]
+        : any;
+};

